docs/devel/releasing.md

```diff
@@ -42,31 +42,67 @@ after the first section.
 Regardless of whether you are cutting a major or minor version, cutting a
 release breaks down into four pieces:
 
-1. Selecting release components.
-1. Tagging and merging the release in Git.
-1. Building and pushing the binaries.
-1. Writing release notes.
+1. selecting release components;
+1. cutting/branching the release;
+1. building and pushing the binaries; and
+1. publishing binaries and release notes.
 
 You should progress in this strict order.
 
-### Building a New Major/Minor Version (`vX.Y.0`)
+### Selecting release components
+
+First, figure out what kind of release you're doing, what branch you're cutting
+from, and other prerequisites.
+
+* Alpha releases (`vX.Y.0-alpha.W`) are cut directly from `master`.
+  * Alpha releases don't require anything besides green tests, (see below).
+* Beta releases (`vX.Y.Z-beta.W`) are cut from their respective release branch,
+  `release-X.Y`.
+  * Make sure all necessary cherry picks have been resolved.  You should ensure
+    that all outstanding cherry picks have been reviewed and merged and the
+    branch validated on Jenkins. See [Cherry Picks](cherry-picks.md) for more
+    information on how to manage cherry picks prior to cutting the release.
+  * Beta releases also require green tests, (see below).
+* Official releases (`vX.Y.Z`) are cut from their respective release branch,
+  `release-X.Y`.
+  * Official releases should be similar or identical to their respective beta
+    releases, so have a look at the cherry picks that have been merged since
+    the beta release and question everything you find.
+  * Official releases also require green tests, (see below).
+* New release series are also cut directly from `master`.
+  * **This is a big deal!**  If you're reading this doc for the first time, you
+    probably shouldn't be doing this release, and should talk to someone on the
+    release team.
+  * New release series cut a new release branch, `release-X.Y`, off of
+    `master`, and also release the first beta in the series, `vX.Y.0-beta`.
```

Comment:
Nope.  When we decide we want to branch v1.2, we'll immediately cut a v1.2.0-beta.0.  Though this should have gotten updated for versioned beta releases.  Done in a9fc43d.